### ISSUE TITLE
paddledetection如何类别加权

### ISSUE THREAD

**Issue by liangruofei** [PaddlePaddle/PaddleDetection#4107]
各位大佬：

       现在我对交通目标训练，7中类别，用ppyolov2训练，由于类别不均衡，检测率和map一直上不去，想paddledetection对类别进行加权以使样本间变得均衡，不知道，paddledetection有没有相关的api或在哪快修改呢

**Comment by wangxinxin08** [comment]
检测率和map上不去，可以先检查下:
1. 学习率是否根据当前的学习率进行了修改
2. 自定义数据集num_classes是否进行了修改

对于类别进行加权的话，可以在https://github.com/PaddlePaddle/PaddleDetection/blob/release/2.2/ppdet/modeling/losses/yolo_loss.py#L109
修改，参考https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/nn/functional/binary_cross_entropy_with_logits_cn.html#binary-cross-entropy-with-logits 设置pos weight属性

**Comment by liangruofei** [comment]
收到，感谢，我先试下


在2021年9月2日 ***@***.***> 写道：

检测率和map上不去，可以先检查下:

学习率是否根据当前的学习率进行了修改
自定义数据集num_classes是否进行了修改

对于类别进行加权的话，可以在https://github.com/PaddlePaddle/PaddleDetection/blob/release/2.2/ppdet/modeling/losses/yolo_loss.py#L109
修改，参考https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/nn/functional/binary_cross_entropy_with_logits_cn.html#binary-cross-entropy-with-logits 设置pos weight属性

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Myxrf** [comment]
> 检测率和map上不去，可以先检查下:
> 
> 1. 学习率是否根据当前的学习率进行了修改
> 2. 自定义数据集num_classes是否进行了修改
> 
> 对于类别进行加权的话，可以在https://github.com/PaddlePaddle/PaddleDetection/blob/release/2.2/ppdet/modeling/losses/yolo_loss.py#L109
> 修改，参考https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/nn/functional/binary_cross_entropy_with_logits_cn.html#binary-cross-entropy-with-logits 设置pos weight属性

你好，我现在是在mask rcnn上训练，样本极不平衡，能请问一下这个网络的话loss加权重需要在哪里改吗，有点不太看得懂
